libraries/standard/mqtt/src/mqtt.c

```diff
@@ -199,6 +203,79 @@ static MQTTStatus_t receiveConnack( MQTTContext_t * const pContext,
                                     MQTTPacketInfo_t * const pIncomingPacket,
                                     bool * const pSessionPresent );
 
+/**
+ * @brief Resends pending acks for a re-established MQTT session.
+ *
+ * @param[in] pContext Initialized MQTT context.
+ *
+ * @return #MQTTSendFailed if transport send failed;
+ * #MQTTSuccess otherwise.
+ */
+static MQTTStatus_t resendPendingAcks( MQTTContext_t * const pContext );
+
+/**
+ * @brief Resends PUBREL for the outgoing QoS2 publish and updates
+ * the state machine based on the state of the packet.
+ *
+ * @param[in] pContext Initialized MQTT context.
+ * @param[in] state Current state of the packet.
+ *
+ * @return #MQTTSendFailed if transport send failed;
+ * #MQTTSuccess otherwise.
+ */
+static MQTTStatus_t resendPubrelForPacketInState( MQTTContext_t * const pContext,
+                                                  MQTTPublishState_t state );
+
+/**
+ * @brief Checks if an incoming publish was received before.
+ *
+ * @param[in] pContext Initialized MQTT context.
+ * @param[in] packetId Packet identifier of the incoming publish packet.
+ * @param[in] qos Qos level of the incoming publish packet.
+ * @param[out] pIsStateUpdateNeededAfterAck Whether a state update is required
+ * after sending an ack for the incoming publish.
+ * @param[out] pPublishRecordState Publish record state for the ack to be sent.
```

Comment:
Don't think this parameter is ever used

Reply:
yea. Thanks. Will remove it.

Reply:
This is used when a MQTT_UpdateStatePublish call is not required to update the incoming publish state.